packages/react/src/switch/Switch.tsx

```diff
@@ -0,0 +1,61 @@
+import * as RadixLabel from "@radix-ui/react-label";
+import * as RadixSwitch from "@radix-ui/react-switch";
+import { type ComponentPropsWithRef, type ElementRef, forwardRef } from "react";
+
+import { Box } from "../box";
+import { Flex } from "../flex";
+import { extractSprinkles } from "../sprinkles";
+import { Text } from "../text";
+import { type ExtendProps } from "../utils";
+import * as styles from "./Switch.css";
+
+type SwitchProps = ExtendProps<
+  ComponentPropsWithRef<typeof RadixSwitch.Root>,
+  ComponentPropsWithRef<typeof Flex>,
+  {
+    label?: string;
+    readonly?: boolean;
+  } & styles.SwitchVariants
+>;
+
+export const Switch = forwardRef<
+  ElementRef<typeof RadixSwitch.Root>,
+  SwitchProps
+>(
+  (
+    { className, disabled, id, label, readonly, size = "default", ...props },
+    ref,
+  ) => {
+    const { restProps, sprinkleProps } = extractSprinkles(props);
+    return (
+      <Flex
+        alignItems="center"
+        className={className}
+        flexDirection="row"
+        gap="0"
+        {...sprinkleProps}
+      >
+        <Box asChild borderColor="transparent" p="0" rounded="2xl">
+          <RadixSwitch.Root
+            ref={ref}
+            {...restProps}
+            {...styles.root({ size })}
+            disabled={disabled || readonly}
+            id={id}
+          >
```

Comment:
restProps and styleProps should go on the box - not on the inner child
```suggestion
        <Box
          asChild
          borderColor="transparent"
          id={id}
          rounded="2xl"
          {...styles.root({ size })}
        >
          <RadixSwitch.Root
            disabled={disabled || readonly}
            ref={ref}
            {...restProps}
          >
```

Reply:
todo: fix the ts types to catch these automatically